business/mesh.go

```diff
@@ -304,34 +311,22 @@ func (in *MeshService) findRemoteKiali(clusterName string, kubeconfig *kubernete
 		return nil
 	}
 
-	// First try: find a remote Kiali in a namespace with the same name
-	// as the one where "this" Kiali is installed. This is under the assumption that
-	// admins will prefer similar configurations across clusters.
-	if len(os.Getenv("ACTIVE_NAMESPACE")) > 0 {
-		kialiInstances = findKialiInNamespace(os.Getenv("ACTIVE_NAMESPACE"), clusterName, clientSet)
-	}
-
-	// Second try: find a remote Kiali in a namespace with the same name
-	// as the local istio-namespace. This is under the assumption that
-	// Kiali may be installed as an add-on.
-	if kialiInstances == nil && conf.IstioNamespace != os.Getenv("ACTIVE_NAMESPACE") {
-		kialiInstances = findKialiInNamespace(conf.IstioNamespace, clusterName, clientSet)
+	// - The operator and the helm charts set this well
+	//   known "app.kubernetes.io/name=kiali" label. It's also present in the
+	//   Istio addon manifest of Kiali.
+	// - We are using the "istio-reader-service-account" to connect to the
+	//   remote cluster. A typical Istio installation gives privileges to
+	//   this SA to list services in a cluster-wide way.
+	services, getSvcErr := clientSet.GetAllServicesByLabels("app.kubernetes.io/name=kiali")
```

Comment:
People using the server-side helm chart can customize this name. I hate that feature, but some community people want it and use it, so we offer it. So you are not guaranteed `app.kubernetes.io/name` is "kiali". See: https://github.com/kiali/helm-charts/blob/master/kiali-server/values.yaml#L1-L2 and https://github.com/kiali/helm-charts/blob/master/kiali-server/templates/_helpers.tpl#L63

Reply:
I will assume there is a reason why "kiali.io/external-url" is an annotation not a label (I know there are limitations of label values such as length and allowed characters, maybe URLs are not available for label values). But if "kiali.io/external-url" is a label not an annotation then you can search on that very label. But, again, I'll assume there is a reason it is an annotation.

Reply:
I just realized there is another label you can use: https://github.com/kiali/helm-charts/blob/master/kiali-server/templates/_helpers.tpl#L56
```
app.kubernetes.io/part-of: kiali
```

We attach this label to indicate this resource is a part of the Kiali ecosystem. This is most likely what you want to use.

Reply:
It apparently works when using `app.kubernetes.io/part-of: kiali`.
I see the label is also present in the Istio add-on, so it shouldn't be an issue.

I changed it.